--- a/canvas-server/build.gradle.kts
+++ b/canvas-server/build.gradle.kts
@@ -56,7 +_,30 @@
         }
     }
 
-    activeFork = canvas
+     val baguette = forks.register("baguette") {
+         forks = canvas
+         upstream.patchRepo("paperServer") {
+             upstreamRepo = canvas.patchedRepo("paperServer")
+             patchesDir = rootDirectory.dir("baguette-server/paper-patches")
+             additionalAts = rootDirectory.file("build-data/baguette-paperserver.at")
+             outputDir = rootDirectory.dir("paper-server")
+        }
+         upstream.patchRepo("foliaServer") {
+             upstreamRepo = canvas.patchedRepo("foliaServer")
+             patchesDir = rootDirectory.dir("baguette-server/folia-patches")
+             additionalAts = rootDirectory.file("build-data/baguette-foliaserver.at")
+             outputDir = rootDirectory.dir("folia-server")
+        }
+        upstream.patchDir("canvasServer") {
+            upstreamPath = "canvas-server"
+            excludes = setOf("src/minecraft", "paper-patches", "folia-patches", "minecraft-patches", "build.gradle.kts", "build.gradle.kts.patch")
+            patchesDir = rootDirectory.dir("baguette-server/canvas-patches")
+            additionalAts = rootDirectory.file("build-data/baguette-canvasserver.at")
+            outputDir = rootDirectory.dir("canvas-server")
+        }
+    }
+
+    activeFork = baguette
 
 
     spigot {
@@ -146,12 +_,16 @@
         resources { srcDir("../paper-server/src/main/resources") }
         java { srcDir("../folia-server/src/main/java") }
         resources { srcDir("../folia-server/src/main/resources") }
+        java { srcDir("../canvas-server/src/main/java") }
+        resources { srcDir("../canvas-server/src/main/resources") }
     }
     test {
         java { srcDir("../paper-server/src/test/java") }
         resources { srcDir("../paper-server/src/test/resources") }
         java { srcDir("../folia-server/src/test/java") }
         resources { srcDir("../folia-server/src/test/resources") }
+        java { srcDir("../canvas-server/src/test/java") }
+        resources { srcDir("../canvas-server/src/test/resources") }
     }
 }
 val log4jPlugins = sourceSets.create("log4jPlugins") {
@@ -178,7 +_,7 @@
 }
 
 dependencies {
-    implementation(projects.canvasApi)
+    implementation(projects.baguetteApi)
     implementation("ca.spottedleaf:concurrentutil:0.0.3")
     implementation("org.jline:jline-terminal-ffm:3.29.0") // use ffm on java 22+
     implementation("org.jline:jline-terminal-jni:3.29.0") // fall back to jni on java 21
@@ -266,14 +_,14 @@
         val gitBranch = git.exec(providers, "rev-parse", "--abbrev-ref", "HEAD").get().trim()
         attributes(
             "Main-Class" to "org.bukkit.craftbukkit.Main",
-            "Implementation-Title" to "Canvas",
+            "Implementation-Title" to "Baguette",
             "Implementation-Version" to implementationVersion,
             "Implementation-Vendor" to date,
-            "Specification-Title" to "Canvas",
+            "Specification-Title" to "Baguette",
             "Specification-Version" to project.version,
-            "Specification-Vendor" to "Canvas Team",
-            "Brand-Id" to "canvasmc:canvas",
-            "Brand-Name" to "Canvas",
+            "Specification-Vendor" to "Baguette Team",
+            "Brand-Id" to "canvasmc:baguette",
+            "Brand-Name" to "Baguette",
             "Build-Number" to (build ?: ""),
             "Build-Time" to buildTime.toString(),
             "Git-Branch" to gitBranch,
@@ -374,8 +_,8 @@
     })
     jvmArgs("--add-modules=jdk.incubator.vector")
 
-    if (rootProject.childProjects["canvas-test-plugin"] != null) {
-        val testPluginJar = rootProject.project(":canvas-test-plugin").tasks.jar.flatMap { it.archiveFile }
+    if (rootProject.childProjects["test-plugin"] != null) {
+        val testPluginJar = rootProject.project(":test-plugin").tasks.jar.flatMap { it.archiveFile }
         inputs.file(testPluginJar)
         args("-add-plugin=${testPluginJar.get().asFile.absolutePath}")
     }
